run-basic-batch-operators-test.sh

```diff
@@ -0,0 +1,208 @@
+	echo "Background PID: $background_pid"
+
+	# get all the packages present in the cluster catalog
+	oc get packagemanifest | grep "Red Hat Operators" | awk '{print $1}' >"$REPORT_FOLDER"/redhat-operators.txt
+
+	# get BundleImage for each of the packages
+	while read -r i; do getLatestBundle "$i"; done <"$REPORT_FOLDER"/redhat-operators.txt >"$BUNDLE_PATH"
+
+	# kill the port forwarding
+	kill -9 "$background_pid"
+	ORIG_BUNDLE_PATH=redhat-operators
+else
+	# if user provided file is present, copy it in the report
+	cp "$ORIG_BUNDLE_PATH" "$BUNDLE_PATH"
+fi
+
+# check for docker config file
+if [ ! -e "$DOCKER_CONFIG" ]; then
+	echo "Docker config is missing at $DOCKER_CONFIG"
+	exit 1
+fi
+
+# check KUBECONFIG
+if [ -z "$KUBECONFIG" ]; then
+	echo "The environment variable KUBECONFIG is not set."
+	exit 1
+fi
+
+# Write config file template
+cat <<EOF >"$CONFIG_YAML_TEMPLATE"
+targetNameSpaces:
+  - name: \$ns
+podsUnderTestLabels:
+  - "test-network-function.com/generic: target"
+operatorsUnderTestLabels:
+  - "test-network-function.com/operator: target" 
+EOF
+
+OPERATOR_PAGE='<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>HTTP Link Example</title>'
+
+# add per test run links
+{
+	# add per operator details link
+	echo "Time: <b>$TIMESTAMP</b>, file: <b>$ORIG_BUNDLE_PATH</b>"
+
+	#add detailed results
+	echo ", detailed results: "'<a href="/'"$REPORT_FOLDER_RELATIVE"'/index.html">'"link"'</a>'
+
+	# add CSV file link
+	echo ", CSV: "
+	echo '<a href="/'"$REPORT_FOLDER_RELATIVE"'/results.csv">'"link"'</a>'
+
+	# add operator bundle list link
+	echo ", bundle list: "
+	echo '<a href="/'"$REPORT_FOLDER_RELATIVE"'/'"$BUNDLE_FILENAME"'">'"link"'</a>'
+
+	# new line
+	echo "<br>"
+} >>"$BASE_DIR"/index.html
+
+echo "$OPERATOR_PAGE" >>"$REPORT_FOLDER"/index.html
+# For each bundle in a provided catalog, this script will install the operator and run the CNF test suite.
+while IFS=, read -r package_name bundle_image; do
+	# Workaround for cleaning operator leftovers, see https://access.redhat.com/solutions/6971276
+	oc delete mutatingwebhookconfigurations controller.devfile.io
+	oc delete validatingwebhookconfigurations controller.devfile.io
+
+	# read package name and bundle image from the text file
+	package_name=$(echo "$package_name" | awk '{$1=$1};1')
+	bundle_image=$(echo "$bundle_image" | awk '{$1=$1};1')
+
+	# create a new namespace for each operator install
+	ns=test-"$package_name"
+	oc new-project "$ns"
+
+	# use operator-sdk binary to install the operator in a custom namespace
+	operator-sdk run bundle "$bundle_image"
+
+	reportDir="$REPORT_FOLDER"/"$package_name"
+
+	# store the results of CNF test in a new directory
+	mkdir -p "$reportDir"
+
+	configYaml="$reportDir"/tnf_config.yml
+
+	# change the targetNameSpace in tng_config file
+	sed "s/\$ns/$ns/" "$CONFIG_YAML_TEMPLATE" >"$configYaml"
+
+	# label everything
```

Comment:
It is not obvious which labels are added. Provide examples in the comment.

Reply:
fixed